#! /bin/sh /usr/share/dpatch/dpatch-run
## 30_unused_libs.dpatch by Kevin Goodsell <kevin-opensource@omegacrash.net>
##
## DP: Hack to remove unused libraries from the link command. This is paired
## DP: with changes to config.mak made in the debian/rules file. Fixing the
## DP: configure script to not add unused libraries would be better, but a
## DP: lot more complicated.

@DPATCH@
diff -urNad mplayer-trunk~/Makefile mplayer-trunk/Makefile
--- mplayer-trunk~/Makefile	2010-02-10 10:31:15.000000000 -0800
+++ mplayer-trunk/Makefile	2010-02-10 10:33:41.000000000 -0800
@@ -850,10 +850,12 @@
 
 mencoder$(EXESUF): $(MENCODER_DEPS)
 mencoder$(EXESUF): EXTRALIBS += $(EXTRALIBS_MENCODER)
+mencoder$(EXESUF): REMOVE_LIBS =
 mplayer$(EXESUF): $(MPLAYER_DEPS)
 mplayer$(EXESUF): EXTRALIBS += $(EXTRALIBS_MPLAYER)
+mplayer$(EXESUF): REMOVE_LIBS = $(MPLAYER_REMOVE_LIBS)
 mencoder$(EXESUF) mplayer$(EXESUF):
-	$(CC) -o $@ $^ $(EXTRALIBS)
+	$(CC) -o $@ $^ $(filter-out $(REMOVE_LIBS),$(EXTRALIBS))
 
 codec-cfg$(EXESUF): codec-cfg.c help_mp.h
 	$(HOST_CC) -O -DCODECS2HTML -I. -o $@ $<
